Mediapipe docs update (#2045)
openvinotoolkit/model_server: docs/mediapipe.md

|pbtxt line|input/output|tag|packet type|stream name|
|:---|:---|:---|:---|:---|
|input_stream: "a"|input|none|ov::Tensor|a|
|input_stream: "IMAGE:a"|input|IMAGE|mediapipe::Image|a|
|output_stream: "OVTENSOR:b"|output|OVTENSOR|ov::Tensor|b|
|input_stream: "REQUEST:req"|input|REQUEST|KServe inference::ModelInferRequest|req|
|output_stream: "RESPONSE:res"|output|RESPONSE|KServe inference::ModelInferResponse|res|

In case of missing tag OVMS assumes that the packet type is `ov::Tensor'.
For list of supported packet types and tags of OpenVINOInferenceCalculator check documentation of [OVMS calculators](https://github.com/openvinotoolkit/model_server/blob/main/src/mediapipe_calculators/calculators.md).

With KServe gRPC API you are also able to push side input packets into graph. In this case created side packet type is the same as KServe parameter type (string, int64 or boolean).

TODO add details about IMAGE @mkulakow

Documentation on handling tags inside OVMS calculators is placed [here](https://github.com/openvinotoolkit/mediapipe/blob/237bf3bf23c4d7b7e38eb92b4a3b6d540d83421b/mediapipe/calculators/ovms/calculators.md).

## Configuration files <a name="configuration-files"></a>

> MediaPipe pipelines configuration is to be placed in the same json file like the 

Reviewer: I think graph is a better word here

Author: changed to servable

```

Here can be found more information about [MediaPipe graphs proto](https://developers.google.com/mediapipe/framework/framework_concepts/graphs)

# Supported input/output packet types

OVMS does support processing several packet types at the inputs and outputs of the graph.
Following table lists supported tag and packet types based on pbtxt configuration file line:

|pbtxt line|input/output|tag|packet type|stream name|
|:---|:---|:---|:---|:---|
|input_stream: "a"|input|none|ov::Tensor|a|
|input_stream: "IMAGE:a"|input|IMAGE|mediapipe::ImageFrame|a|
|output_stream: "OVTENSOR:b"|output|OVTENSOR|ov::Tensor|b|
|input_stream: "REQUEST:req"|input|REQUEST|KServe inference::ModelInferRequest|req|
|output_stream: "RESPONSE:res"|output|RESPONSE|KServe inference::ModelInferResponse|res|

In case of missing tag OVMS assumes that the packet type is `ov::Tensor'.

> For list of supported packet types and tags of OpenVINOInferenceCalculator check documentation of [OVMS calculators](https://github.com/openvinotoolkit/model_server/blob/main/src/mediapipe_calculators/calculators.md).

Reviewer 2: Refering to non existing file

With KServe gRPC API you are also able to push side input packets into graph. In this case created side packet type is the same as KServe parameter type (string, int64 or boolean).

`Image` inputs requires image pixel data inside raw_input_contents that can be converted to Mediapipe::ImageFrame format. For now, those kind of inputs only accepts three-dimensional data in HWC layout. Datatypes supported for `Image` format:
|Datatype|Number of channels|
|:---|:---|
|FP16|1,3,4|
|FP32|1,2|
|UINT8|1,3,4|
|INT8|1,3,4|
|UINT16|1,3,4|
|INT16|1,3,4|

> Documentation on handling tags inside OVMS calculators is placed [here](https://github.com/openvinotoolkit/mediapipe/blob/237bf3bf23c4d7b7e38eb92b4a3b6d540d83421b/mediapipe/calculators/ovms/calculators.md).

Reviewer 2: Refering to non existing file

# Supported input/output packet types

OVMS does support processing several packet types at the inputs and outputs of the graph.
Following table lists supported tag and packet types based on pbtxt configuration file line:

|pbtxt line|input/output|tag|packet type|stream name|
|:---|:---|:---|:---|:---|
|input_stream: "a"|input|none|ov::Tensor|a|
|input_stream: "IMAGE:a"|input|IMAGE|mediapipe::ImageFrame|a|
|output_stream: "OVTENSOR:b"|output|OVTENSOR|ov::Tensor|b|
|input_stream: "REQUEST:req"|input|REQUEST|KServe inference::ModelInferRequest|req|
|output_stream: "RESPONSE:res"|output|RESPONSE|KServe inference::ModelInferResponse|res|

In case of missing tag OVMS assumes that the packet type is `ov::Tensor'.
For list of supported packet types and tags of OpenVINOInferenceCalculator check documentation of [OVMS calculators](https://github.com/openvinotoolkit/model_server/blob/main/src/mediapipe_calculators/calculators.md).

With KServe gRPC API you are also able to push side input packets into graph. In this case created side packet type is the same as KServe parameter type (string, int64 or boolean).

> `Image` inputs requires image pixel data inside raw_input_contents that can be converted to MediaPipe ImageFrame format. For now, those kind of inputs only accepts three-dimensional data in HWC layout. Datatypes supported for `Image` format:

Reviewer: ```suggestion
`Image` inputs requires image pixel data inside `raw_input_contents` that can be converted to MediaPipe ImageFrame format. For now, those kind of inputs only accepts three-dimensional data in HWC layout. Datatypes supported for `Image` format:
```

> |Datatype|Number of channels|

Reviewer: ```suggestion
|Datatype|Allowed number of channels|
```

|:---|:---|
|FP16|1,3,4|
|FP32|1,2|
|UINT8|1,3,4|
|INT8|1,3,4|
|UINT16|1,3,4|
|INT16|1,3,4|

> Documentation on handling tags inside OVMS calculators is placed [here](https://github.com/openvinotoolkit/mediapipe/blob/237bf3bf23c4d7b7e38eb92b4a3b6d540d83421b/mediapipe/calculators/ovms/calculators.md).

Reviewer: ```suggestion
Check the documentation [handling tags inside OpenVINO Model Server calculators]((https://github.com/openvinotoolkit/mediapipe/blob/main/mediapipe/calculators/ovms/calculators.md).

Review also an example in [object detection demo](../demos/mediapipe/object_detection/README.md)
```

{
    "model_config_list": [...],
    "mediapipe_config_list": [
        {
            "name":"mediaDummy",
            "base_path":"/mediapipe/graphs/",
            "graph_path":"graphdummyadapterfull.pbtxt",
            "subconfig":"subconfig_dummy.json"

>     }

Reviewer 2: indent

and [REST Model Status](model_server_rest_api_kfs.md)

## MediaPipe Graphs Examples <a name="graphs-examples"></a>

[Image classification](../demos/mediapipe/image_classification/README.md)

[Multi model](../demos/mediapipe/multi_model_graph/README.md)

## Current limitations <a name="current-limitations"></a>
- MediaPipe graphs are supported only for gRPC KServe API.

- KServe ModelMetadata call response contains only input and output names. In the response shapes will be empty and datatypes will be `"INVALID"`.

- Binary inputs are not supported for MediaPipe graphs.

- Making changes in subconfig file does not trigger config reloads. Main config changes are monitored and triggers subconfig reload even if this wasn't changed. Changes in main config json trigger also checking for changes in graph's pbtxt files.

## Adding your own mediapipe calculator to OVMS <a name="adding-calculator"></a>
If you want ot add your own mediapipe calculator to OVMS functionality you need to add it as a dependency and rebuild the ovms binary.

If you have it in external repository, you need to add the http_archive() definition or git_repository() definition to the bazel WORKSPACE file.

> Then you need to add the calculator target as a bazel dependency to the src/BUILD file. This should be done for:

Reviewer: ```suggestion
Then you need to add the calculator target as a bazel dependency to the [src/BUILD](../src/BUILD) file. This should be done for:
```

Graphs can be queried for their state using the calls [GetModelStatus](model_server_grpc_api_kfs.md)
and [REST Model Status](model_server_rest_api_kfs.md)

## MediaPipe Graphs Examples <a name="graphs-examples"></a>

[Image classification](../demos/mediapipe/image_classification/README.md)

[Multi model](../demos/mediapipe/multi_model_graph/README.md)

## Current limitations <a name="current-limitations"></a>

> - Mediapipe graphs are supported only for gRPC KServe API. Only Tensorflow Servin API calls supported are get model status and config reload.

Reviewer: config/reload is not a part of TFS API
